src/components/ChatEntry.js

```diff
@@ -1,22 +1,40 @@
 import React from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
+import TimeStamp from './TimeStamp';
 
-const ChatEntry = (props) => {
+import { useState } from 'react';
+
+const ChatEntry = ({id, sender, body, timeStamp, updateLike}) => {
+  const [changeHeart, setChangeHeart] = useState('🤍');
+  const fillHeart = () => (changeHeart === '🤍' ? setChangeHeart('❤️') : setChangeHeart('🤍'));
+  
   return (
     <div className="chat-entry local">
-      <h2 className="entry-name">Replace with name of sender</h2>
+      <h2 className="entry-name">{sender}</h2>
       <section className="entry-bubble">
-        <p>Replace with body of ChatEntry</p>
-        <p className="entry-time">Replace with TimeStamp component</p>
-        <button className="like">🤍</button>
+        <p>{body}</p>
+        <p className="entry-time">
+          <TimeStamp time={timeStamp} />
+        </p>
+        <button className="like" onClick={() => {
+            updateLike({id, liked: !changeHeart});
+            fillHeart();
+          }}
+          >{changeHeart}
+        </button>
       </section>
     </div>
   );
 };
 
 ChatEntry.propTypes = {
-  //Fill with correct proptypes
+  id: PropTypes.number.isRequired,
+  sender: PropTypes.string.isRequired, 
+  body: PropTypes.string.isRequired,
+  timeStamp: PropTypes.string.isRequired,
+  liked: PropTypes.bool,
+  updateLike: PropTypes.func
```

Comment:
I'd recommend making the `updateLike` prop required since it is essential for the like functionality to work properly.